test/test_generators.py

```diff
@@ -25,3 +32,186 @@ def test_default_param_filler():
         'timeout_sec': 5,
         'headers': {}
     }
+
+
+# Tests for batching functionality
```

Comment:
Probably went a little overboard here, but I like to be thorough...  😬

Reply:
omg this is awesome!  <3 overboard tests